src/main/java/com/werfen/report/model/GridPageDataSource.java

```diff
@@ -0,0 +1,13 @@
+package com.werfen.report.model;
+
+import java.util.List;
+
+public interface GridPageDataSource {
+    void moveFirst();
+
+    boolean nextPage();
+
+    int getRowCount();
```

Comment:
After deeper analysis with this code, I don't see the point on having a getRowCount() method on the data source. That would make harder the implementation of the page data source when it is possible to iterate through all the rows of the data source without a global count. Moreover, every call to this row count may imply an extra query to the source repository, though it can be inefficient.